src/main/java/org/terasology/climateConditions/ExtremeClimateConditionSystem.java

```diff
@@ -0,0 +1,150 @@
+import org.terasology.logic.health.event.ActivateRegenEvent;
+import org.terasology.logic.health.event.DeactivateRegenEvent;
+import org.terasology.logic.health.event.DoDamageEvent;
+import org.terasology.logic.location.Location;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.physics.events.ImpulseEvent;
+import org.terasology.physics.events.MovedEvent;
+import org.terasology.registry.In;
+import org.terasology.thirst.ThirstUtils;
+import org.terasology.thirst.component.ThirstComponent;
+import org.terasology.utilities.random.Random;
+
+import static org.terasology.logic.health.RegenAuthoritySystem.BASE_REGEN;
+import static org.terasology.thirst.ThirstAuthoritySystem.THIRST_DAMAGE_ACTION_ID;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class ExtremeClimateConditionSystem extends BaseComponentSystem{
+    public static final String FROSTBITE_DAMAGE_ACTION_ID = "Frostbite Damage";
+
+    @In
+    private EntityManager entityManager;
+    @In
+    private DelayManager delayManager;
+    @In
+    private PrefabManager prefabManager;
+    @In
+    private Time time;
+
+    private int healthDecreaseInterval = 20000;
+    private int healthDecreaseAmount = 10;
+    private float thresholdHeight= (float)80.0;
+    private Random random;
+    /*
+    NOTE: This part is for the desert effects and is incomplete.
+
+    @ReceiveEvent(components = {ThirstComponent.class, CharacterMovementComponent.class})
+    public void EnterDesert(OnPlayerSpawnedEvent event, EntityRef character, ThirstComponent thirst, CharacterMovementComponent movement) {
+
+        /*
+        Biome newBiome= event.getNewBiome();
+        if(newBiome.getName() == "Desert")
+        {
+            thirst.sprintDecayPerSecond*=10;
+            thirst.normalDecayPerSecond*=10;
+        }
+
+        thirst.sprintDecayPerSecond*=10;
+        thirst.normalDecayPerSecond*=10;
+        movement.runFactor = 1;
+        movement.speedMultiplier = (float)0.8;
+        movement.jumpSpeed = (float)8;
+    }
+*/
+
+    //The following part is for the Snow effects.
+    @ReceiveEvent(components = {PlayerCharacterComponent.class, ThirstComponent.class, CharacterMovementComponent.class})
+    public void ChangeSnowEffect(MovedEvent event, EntityRef player, LocationComponent location, ThirstComponent thirst, CharacterMovementComponent movement) {
+        float height = event.getPosition().getY();
+        float deltaHeight = event.getDelta().getY();
+        float lastHeight = height - deltaHeight;
+        if(height > thresholdHeight && lastHeight <= thresholdHeight) {
+                delayManager.addPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID, 0, healthDecreaseInterval);
+                movement.runFactor = 1;
+                movement.speedMultiplier = (float)0.7;
+                movement.jumpSpeed = (float)8;
```

Comment:
I believe this should be done using events instead of hard-coding the values. IIRC @skaldarnar recently introduced events for changing the player's height that you could check out as a reference.

Reply:
I have tried something but I think some operations are not completely accurate on float datatype so I have created a few extra variables too. Please take a look.